eth-bridge/contracts/src/Bridge.sol

```diff
@@ -0,0 +1,532 @@
+    }
+
+    /// Set the minting delay
+    /// @param delay New minting delay
+    /// @dev Only addresses with SUPER_ADMIN_ROLE can call this
+    function setMintDelay(uint256 delay) public onlyRole(SUPER_ADMIN_ROLE) {
+        mintDelay = delay;
+    }
+
+    /// Claim pending reward
+    /// Used by relays to get ether collected from users as minting fees.
+    /// @dev May revert with EthTransferFailed
+    function claimReward() public {
+        uint256 amount = pendingRewards[msg.sender];
+        pendingRewards[msg.sender] = 0;
+        // disabling check as this is the recommended way of transferring ether
+        // https://solidity-by-example.org/sending-ether/
+        // slither-disable-next-line low-level-calls
+        (bool sent,) = msg.sender.call{value: amount}("");
+        if (!sent) revert EthTransferFailed();
+    }
+
+    /// Set rate limit parameters
+    /// @param counterLimit New counter limit (max burst mint)
+    /// @param decayRate New decay rate (avg mint per block)
+    /// @dev Only addresses with SUPER_ADMIN_ROLE can call this
+    function setRateLimit(uint256 counterLimit, uint256 decayRate) public onlyRole(SUPER_ADMIN_ROLE) {
+        rateLimit.counterLimit = counterLimit;
+        rateLimit.decayRate = decayRate;
+    }
+
+    /// Set max circulating token supply
+    /// @param supplyLimit_ new supply limit
+    /// @dev Only addresses with SUPER_ADMIN_ROLE can call this
+    function setSupplyLimit(uint256 supplyLimit_) public onlyRole(SUPER_ADMIN_ROLE) {
+        supplyLimit = supplyLimit_;
+    }
+
+    /// Transfer ownership of underlying token contract.
+    /// Will stop the bridge and set the token address to 0, effectively
+    /// bricking the bridge.
+    /// @param newOwner new owner
+    /// @dev Only addresses with SUPER_ADMIN_ROLE can call this
+    /// @dev Interacts with `token` contract
+    function transferTokenOwnership(address newOwner) public onlyRole(SUPER_ADMIN_ROLE) {
+        WrappedToken token_ = token;
+        token = WrappedToken(address(0));
+        _setActive(false);
+        token_.transferOwnership(newOwner);
+    }
+
+    /// Check if given receiptId was already voted on by given voter
+    /// @param receiptId ReceiptId to check
+    /// @param voter Voter to check
+    /// @return voted_ true if voter voted for receipt
+    function voted(bytes32 receiptId, address voter) public view returns (bool voted_) {
+        return _arrayContains(votes[receiptId], voter);
+    }
+
+    // solhint-disable-next-line no-empty-blocks
+    function _authorizeUpgrade(address newImplementation) internal override onlyRole(UPGRADER_ROLE) {}
+
+    function _setActive(bool active) internal {
+        if (active && address(token) == address(0)) revert InvalidConfiguration();
+        if (active != bridgeActive) emit StateChanged(active);
+        bridgeActive = active;
+
+        if (address(token) != address(0)) {
+            bool tokenPaused = token.paused();
```

Comment:
Token is the contract of our wrapped LLM or LLD. It's implemented in the `WrappedToken.sol` file. It can be paused, which stops all token transfers. Bridge is stopped if some attack happens - in that case we also stop token transfers to stop attacker from trading whatever he stole.